### PR TITLE
feat: add allocations and revenue forecast to projects page in new admin

### PR DESCRIPTION
**Pull request by thiagofportella**
Closes https://github.com/Codeminer42/Punchclock/issues/518

- Fix tabButtons bug in navbar.js, adding a optional chaining in the script.
- Add allocations and revenue forecast informations for a project in the New Admin show page.
- Create separate resource for allocate users in a project.

### Project show page:
Captura de Tela 2023-08-04 às 17 50 18
Captura de Tela 2023-08-04 às 17 50 37

### Allocate User in Project page:
Captura de Tela 2023-08-04 às 17 59 52
Captura de Tela 2023-08-04 às 17 59 38

### CODE REVIEW

File: app/assets/javascripts/new_admin/navbar.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ tabButtons.forEach((buttonElement) => {
     setActiveButton(buttonElement)
   });
 });
-tabButtons[0].click();
+tabButtons[0]?.click();
```
Comment (adbatista) [inline app/assets/javascripts/new_admin/navbar.js:54]: when the tab buttons will be null?

File: app/views/new_admin/projects/show.html.erb [hunk 2]
```diff
@@ -21,7 +21,29 @@
...
+      <div id="project_allocations" class="my-4">
+        <%= render 'project_allocations', allocations: @allocations %>
+      </div>
+    <% end %>
+
+    <% if @revenue_forecast.any? %>
+      <div id="project_revenue_forecast" class="my-4">
+        <%= render 'project_revenue_forecast', revenue_forecast: @revenue_forecast %>
+      </div>
+    <% end %>
```
Comment (adbatista) [inline app/views/new_admin/projects/show.html.erb:47]: you could use a render collection in those two

Comment (adbatista) [inline app/views/new_admin/projects/show.html.erb:47]: you could use a render collection in those two

File: app/assets/javascripts/new_admin/navbar.js (hunk 1 above)
Comment (thiagofportella) [reply app/assets/javascripts/new_admin/navbar.js:54]: It seems that `Cannot read properties of undefined` is raised when no `data-tab-id` is found in a NewAdmin page.
The following example is from the `show` Project page, when no revenue forecast or allocation is found for the specific project, meaning, in this scenario, no `data-tab-id` (or `data-tab-content`):
Captura de Tela 2023-08-07 às 10 47 32

**Pull request merged**